src/store/modules/util/actions.ts

```diff
@@ -47,6 +49,254 @@ const actions: ActionTree<UtilState, RootState> = {
   },
   async setProductIdentifications({ commit }, payload) {
     commit(types.UTIL_PRODUCT_IDENT_UPDATED, payload)
+  },
+
+  async getForceScanSetting({ commit, dispatch }, eComStoreId) {
+    const payload = {
+      "inputFields": {
+        "productStoreId": eComStoreId,
+        "settingTypeEnumId": "RECEIVE_FORCE_SCAN"
+      },
+      "filterByDate": 'Y',
+      "entityName": "ProductStoreSetting",
+      "fieldList": ["settingValue", "fromDate"],
+      "viewSize": 1
+    }
+
+    try {
+      const resp = await UtilService.getProductStoreSetting(payload) as any
+      if(!hasError(resp)) {
+        const respValue = resp.data.docs[0].settingValue
+        commit(types.UTIL_FORCE_SCAN_STATUS_UPDATED, respValue === "true")
```

Comment:
We need to update the force scan status on error case as well, with its default value. Same applies to barcodePref.